### PR TITLE
UserDevfile API permissions

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
UserDevfile API permissions
Add permission to https://github.com/eclipse/che/pull/17009

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16981

### CODE REVIEW

File: multiuser/integration-tests/che-multiuser-cascade-removal/src/test/java/org/eclipse/che/multiuser/integration/jpa/cascaderemoval/JpaEntitiesCascadeRemovalTest.java [hunk 1]
```diff
@@ -453,6 +484,12 @@ private void createTestData()
     organizationResourcesDistributor.capResources(
         childOrganization.getId(),
         singletonList(new ResourceImpl(RamResourceType.ID, 1024, RamResourceType.UNIT)));
+
+    userDevfileDao.create(devfile = TestObjectsFactory.createUserDevfile("id-dev1", "devfile1"));
```
Comment (sleshchenko) [inline multiuser/integration-tests/che-multiuser-cascade-removal/src/test/java/org/eclipse/che/multiuser/integration/jpa/cascaderemoval/JpaEntitiesCascadeRemovalTest.java:488]: I suppose it's meant to be:
```suggestion
        prepareCreator(user.getId());
        userDevfileDao.create(devfile = TestObjectsFactory.createUserDevfile("id-dev1", "devfile1"));
```

File: multiuser/sql-schema/src/main/resources/che-schema/7.19.0/1.1__add_userdevfile_permissions.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+CREATE TABLE che_userdevfile_permissions_actions (
+  userdevfile_permissions_id       VARCHAR(255),
+  actions         VARCHAR(255)
+);
+-- indexes
+CREATE INDEX che_index_userdevfile_permissions_actions_actions ON che_userdevfile_permissions_actions (actions);
+CREATE INDEX che_index_userdevfile_permissions_actions_userdevfile_id ON che_userdevfile_permissions_actions (userdevfile_permissions_id);
+-- constraints
+ALTER TABLE che_userdevfile_permissions_actions ADD CONSTRAINT che_fk_userdevfile_permissions_actions_id FOREIGN KEY (userdevfile_permissions_id) REFERENCES che_userdevfile_permissions(id);
+--------------------------------------------------------------------------------
```
Comment (sleshchenko) [inline multiuser/sql-schema/src/main/resources/che-schema/7.19.0/1.1__add_userdevfile_permissions.sql:40]:
```suggestion
--------------------------------------------------------------------------------

```

File: multiuser/sql-schema/src/main/resources/che-schema/7.19.0/1.1__add_userdevfile_permissions.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+-- indexes
+CREATE UNIQUE INDEX che_index_userdevfile_permissions_user_id_userdevfile_id ON che_userdevfile_permissions (user_id, userdevfile_id);
+CREATE INDEX che_index_userdevfile_permissions_userdevfile_id ON che_userdevfile_permissions (userdevfile_id);
+-- constraints
+ALTER TABLE che_userdevfile_permissions ADD CONSTRAINT che_fk_userdevfile_permissions_user_id FOREIGN KEY (user_id) REFERENCES usr (id);
+ALTER TABLE che_userdevfile_permissions ADD CONSTRAINT che_fk_userdevfile_permissions_workspace_id FOREIGN KEY (userdevfile_id) REFERENCES userdevfile (id);
+--------------------------------------------------------------------------------
+
+
+-- Worker actions --------------------------------------------------------------
```
Comment (sleshchenko) [inline multiuser/sql-schema/src/main/resources/che-schema/7.19.0/1.1__add_userdevfile_permissions.sql:30]: that's not worker

File: multiuser/permission/che-multiuser-permission-userdevfile/src/main/java/org/eclipse/che/multiuser/permission/devfile/server/UserDevfileDomain.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.List;
+import org.eclipse.che.multiuser.api.permission.server.AbstractPermissionsDomain;
+import org.eclipse.che.multiuser.permission.devfile.server.model.impl.UserDevfilePermissionImpl;
+
+/** Domain for storing devfile's permissions */
+public class UserDevfileDomain extends AbstractPermissionsDomain<UserDevfilePermissionImpl> {
+  public static final String READ = "read";
+  public static final String DELETE = "delete";
+  public static final String UPDATE = "update";
+  public static final String DOMAIN_ID = "userDevfile";
```
Comment (sleshchenko) [inline multiuser/permission/che-multiuser-permission-userdevfile/src/main/java/org/eclipse/che/multiuser/permission/devfile/server/UserDevfileDomain.java:24]: that's visible thing thought API and as we agreed - use just `devfile` for visible things
```suggestion
  public static final String DOMAIN_ID = "devfile";
```

**Comment by skabashnyuk** [comment]
closing in favor https://github.com/eclipse/che/pull/17843

**Comment by skabashnyuk** [comment]
closing in favor https://github.com/eclipse/che/pull/17843

**Pull request closed without merging**